File: test-fixtures/cross-language/scala/realistic.scala
// SPDX-License-Identifier: MIT
package crosslanguage

case class SessionState(var mutableField: String, immutableField: String) {
  private var privateField: Boolean = false
  var sharedName: Int = 0
}

object SessionState {
  val staticField: Int = 1
}

class AuditState {
  var sharedName: Int = 0
}

File: test-fixtures/cross-language/typescript/realistic.ts
// SPDX-License-Identifier: MIT
export class SessionState<T> {
  public mutableField: T;
  public readonly immutableField: string = "fixed";
  public static staticField: number = 0;
  private privateField: boolean = false;
  public sharedName: number = 0;

  constructor(public promotedField: T) {
    this.mutableField = promotedField;
  }
}

export class AuditState {
  public sharedName: number = 0;
}
